test/integration_tests/executor_test.go

```diff
@@ -0,0 +1,25 @@
+package integration_tests
+
+import (
+	"testing"
+
+	"github.com/conductor-sdk/conductor-go/sdk/client"
+	"github.com/conductor-sdk/conductor-go/test/testdata"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetryNotFound(t *testing.T) {
+	executor := testdata.WorkflowExecutor
+	// Workflow id is hardcoded on purpose. It should not be found.
+	err := executor.Retry("2b3ea839-9aeb-11ef-9ac5-ce590b39fb93", true)
+	assert.Error(t, err, "Retry is expected to return an error")
+
+	if swaggerErr, ok := err.(client.GenericSwaggerError); ok {
+		// hmm... this should be a 404 or 400, but it's a 500 right now.
+		assert.Error(t, err, "GetWorkflow was expected to return a 500 error")
```

Comment:
@v1r3n @c4lm - should we fix this backend reponse?

Reply:
https://github.com/orkes-io/orkes-conductor/pull/2167/files